### PR TITLE
[94X] L1 no hack conditions

### PR DESCRIPTION
**Pull request by rekovic**
This is to test PR #20451 plus changes needed in AddOn test.

### CODE REVIEW

**Comment by rekovic** [comment]
please test

**Comment by rekovic** [comment]
please test

**Comment by rekovic** [comment]
I guess I need to do code-check patch first.

**Comment by davidlange6** [comment]
please test

**Comment by kkotov** [comment]
@rekovic 
The last couple of failing tests need L1TCaloConfig. My impression is that it was yet anther general solution for the problem that never existed and we ended up not using it for anything. Looking closer I see that it indeed is a dummy payload. I propose removing it from the code rather than pushing it to the GTs.

**Comment by arunhep** [comment]
@rekovic 
this file is not needed : https://github.com/cms-sw/cmssw/pull/20577/files#diff-8cbfe6480e6018966eac475523946acd as master is already at what you want.

and also this one : https://github.com/cms-sw/cmssw/pull/20577/files#diff-0e32bec8fef95e9e8678732f08ad0e59
because here also master is already at what you want to change.

**Comment by kkotov** [comment]
@rekovic , The solution for the last couple of failing AddOns can be cherry-picked from my ce524f9

**Comment by kkotov** [comment]
@rekovic , ping (cherry-pick ce524f9)

**Comment by rekovic** [comment]
please test

**Comment by Martin-Grunewald** [comment]
+1

**Comment by Martin-Grunewald** [comment]
This PR now has merge conflicts! Please fix!

File: Configuration/AlCa/python/autoCond.py [hunk 1]
```diff
@@ -10,17 +10,19 @@
     # GlobalTag for MC production (p-Pb collisions) with realistic alignment and calibrations for Run1
     'run1_mc_pa'        :   '93X_mcRun1_pA_v0',
     # GlobalTag for MC production with perfectly aligned and calibrated detector for Run2
-    'run2_design'       :   '93X_mcRun2_design_v0',
+    'run2_design'       :   '93X_mcRun2_design_v1',
```
Comment (arunhep) [inline Configuration/AlCa/python/autoCond.py]: @rekovic you need to rebase this to master

Comment (arunhep) [reply Configuration/AlCa/python/autoCond.py]: or rather, you don't need this file in your PR. autocond from master should suffice. Please remove this.

**Comment by rekovic** [comment]
please test

**Comment by rekovic** [comment]
+1

**Comment by rekovic** [comment]
@smuzaffar 
Failure of unit tests, should have nothing to do with this PR, but ```PATIsolatedTrackProducer```, as shown in excerpt from the unit-test report below.

Is this fixed by now?

```
%MSG
----- Begin Fatal Exception 02-Oct-2017 20:13:04 CEST-----------------------
An exception of category 'InvalidReference' occurred while
   [0] Processing  Event run: 274199 lumi: 125 event: 245779397 stream: 0
   [1] Running path 'MINIAODSIMoutput_step'
   [2] Prefetching for module PoolOutputModule/'MINIAODSIMoutput'
   [3] Calling method for module PATIsolatedTrackProducer/'isolatedTracks'
Exception Message:
ValueMap: no associated value for given product and index
----- End Fatal Exception -------------------------------------------------
02-Oct-2017 20:13:04 CEST  Closed file file:simulated_and_cleaned.root

=============================================

MessageLogger Summary

 type     category        sev    module        subroutine        count    total
 ---- -------------------- -- ---------------- ----------------  -----    -----
    1 ConvertedPhotonProdu -e ConvertedPhotonP                       2        2
    2 MissingInput         -e ReducedRecHitCol                       1        1
    3 MissingInput         -e ReducedRecHitCol                       1        1
    4 TkDetLayers          -e ConversionTrackC                       1        1
    5 triggerEventValid    -e PATTriggerProduc                       1        1
    6 Fatal Exception      -s PostProcessPath                        1        1
    7 fileAction           -s file_close                             1        1
    8 fileAction           -s file_open                              2        2

 type    category    Examples: run/evt        run/evt          run/evt
 ---- -------------------- ---------------- ---------------- ----------------
    1 ConvertedPhotonProducer 274199/245779397 274199/245779397 
    2 MissingInput         274199/245779397                  
    3 MissingInput         274199/245779397                  
    4 TkDetLayers          Run: 274199 Stre                  
    5 triggerEventValid    274199/245779397                  
    6 Fatal Exception      274199/245779397                  
    7 fileAction           PostGlobalEndRun                  
    8 fileAction           pre-events       pre-events       

Severity    # Occurrences   Total Occurrences
--------    -------------   -----------------
Error                   6                   6
System                  4                   4

dropped waiting message count 0
+ die 'Failure during merging step' 76
+ echo Failure during merging step: status 76
Failure during merging step: status 76
+ exit 76
status = 19456

---> test testTauEmbeddingProducers had ERRORS
```

**Comment by smuzaffar** [comment]
please test
lets rerun the tests as previous one picked up extra commits which re-build 80% of the IB

**Comment by rekovic** [comment]
@Dr15Jones, @vazzolini, @kmaeshima, @dmitrijus, @franzoni, @vanbesien 
The emerged conflict was resolved and tests pass. Please review/sign.

**Comment by dmitrijus** [comment]
+1

**Comment by Dr15Jones** [comment]
+1
just configuration changes for which the knowledgeable L2 has already signed.

**Comment by davidlange6** [comment]
hi @rekovic - are the changes seen in DQM expected?

**Comment by rekovic** [comment]
@davidlange6 Looking into it.

**Comment by Martin-Grunewald** [comment]
please test

**Comment by Martin-Grunewald** [comment]
@kkotov @rekovic 

Same error, now that #20286 is integrated:
```
----- Begin Fatal Exception 09-Oct-2017 10:54:15 CEST-----------------------
An exception of category 'NoRecord' occurred while
   [0] Processing  Event run: 1 lumi: 1 event: 1 stream: 0
   [1] Running path 'L1simulation_step'
   [2] Calling method for module L1TMuonEndCapTrackProducer/'simEmtfDigis'
Exception Message:
No "L1TMuonEndCapParamsRcd" record found in the EventSetup for synchronization value
Run: 1 LuminosityBlock: 1 Event: 0 Time: 1
 Please add an ESSource or ESProducer that delivers such a record.
----- End Fatal Exception -------------------------------------------------
```
The above error usually points to the payload being in the DB, but not for the run in question!
This real-data test is running on old(er) 2017 data.

**Comment by rekovic** [comment]
@Martin-Grunewald  This is understood.  I will provide a fix soon. 
EMTF has to use the hackConditions (which are not real hackConditions
as the params are obtained from the DB via Frontier, and not yet available via GT.

**Comment by rekovic** [comment]
please test

**Comment by kkotov** [comment]
@Martin-Grunewald , @abrinke1 
According to the summary this error also shows up in the MC workflows (e.g. in 20434.0), but it runs just fine in my local environment with CMSSW_9_4_X_2017-10-08-2300. I observe the same in the data workflows. I will catch up with @rekovic after lunch and see if we can reproduce it.

**Comment by perrotta** [comment]
-1
please rebase

**Comment by rekovic** [comment]
please test

**Comment by rekovic** [comment]
please test

**Comment by rekovic** [comment]
@smuzaffar code-checks are with files that are not part of the rebased version of this PR.
It looks like github is confused and these are left over from before rebase.
Can we restart the `code-checks` ?

**Comment by davidlange6** [comment]
code-checks

**Comment by davidlange6** [comment]
@rekovic - I believe you can do it as well...

**Comment by smuzaffar** [comment]
yes, any one should be able to trigger it

**Comment by rekovic** [comment]
@smuzaffar @davidlange6 Thx.

**Comment by rekovic** [comment]
needs rebase

**Comment by rekovic** [comment]
-1

**Comment by rekovic** [comment]
+1

**Comment by Martin-Grunewald** [comment]
please test

**Comment by rekovic** [comment]
superseded by https://github.com/cms-sw/cmssw/pull/20871

**Pull request closed without merging**